src/rust/ide/lib/view/src/graph_editor/component/documentation_view.rs

```diff
@@ -0,0 +1,585 @@
+        let logger  = Logger::new("DocumentationView");
+        let div     = web::create_div();
+        let dom     = DomSymbol::new(&div);
+        let screen = scene.camera().screen();
+        let size    = Rc::new(Cell::new(Vector2(290.0,screen.height - 30.0)));
+
+        dom.dom().set_style_or_warn("white-space"     ,"normal"                             ,&logger);
+        dom.dom().set_style_or_warn("overflow-y"      ,"auto"                               ,&logger);
+        dom.dom().set_style_or_warn("overflow-x"      ,"auto"                               ,&logger);
+        dom.dom().set_style_or_warn("background-color","rgba(255, 255, 255, 0.85)"          ,&logger);
+        dom.dom().set_style_or_warn("padding"         ,"5px"                                ,&logger);
+        dom.dom().set_style_or_warn("pointer-events"  ,"auto"                               ,&logger);
+        dom.dom().set_style_or_warn("border-radius"   ,"14px"                               ,&logger);
+        dom.dom().set_style_or_warn("width"           ,format!("{}px", 290)                 ,&logger);
+        dom.dom().set_style_or_warn("height"          ,format!("{}px", screen.height - 30.0),&logger);
+
+        scene.dom.layers.main.manage(&dom);
+        DocumentationViewModel{dom,logger,size}.init()
+    }
+
+    fn init(self) -> Self {
+        self.reload_style();
+        self
+    }
+
+    fn set_size(&self, size:Vector2) {
+        self.size.set(size);
+        self.reload_style();
+    }
+
+    fn receive_data(&self, data:&Data) -> Result<(),DataError> {
+        let data_inner = match data {
+            Data::Json {content} => content,
+            _ => todo!()
+        };
+        let placeholder_str = "<h3>Enso Documentation Viewer</h3><p>No documentation available</p>";
+
+        let data_str = serde_json::to_string_pretty(&**data_inner);
+        let data_str = data_str.unwrap_or_else(|e| format!("<Cannot render data: {}>", e));
+        // Fixes a Doc Parser Bug - to be removed when rewritten to rust
+        let data_str = data_str.replace("\\n", "\n");
+        let data_str = data_str.replace("\"", "");
+
+        let parser = parser::DocParser::new_or_panic();
+        let output = parser.generate_html_doc_pure(data_str);
+        let output = output.unwrap_or_else(|_| String::from(placeholder_str));
+        // Fixes a Doc Parser related idea, where stylesheet was a separate file
+        let output = output.replace(r#"<link rel="stylesheet" href="style.css" />"#, "");
+
+        let data_str = format!(r#"<div class="docVis">{}{}</div>"#,
+                               get_doc_style(), output);
+        self.dom.dom().set_inner_html(&data_str);
+        Ok(())
+    }
+
+    /// Generates welcome screen HTML.
+    pub fn welcome_screen(&self) {
+        let data_str = format!(r#"<div class="docVis">{}<h3>Enso Documentation Viewer</h3>
+        <p>No documentation available</p></div>"#, get_doc_style());
```

Comment:
This constant is repeated.

Additionally, I don't think "No documentation available" is a good message here, because I suggest that it is not available at all - and perhaps we just wait for it.

Reply:
Ok, I've been kinda inspired by Xcode, Where it just says `no selection` or `no editor` etc.
<img width="1512" alt="Zrzut ekranu 2020-08-4 o 10 53 03" src="https://user-images.githubusercontent.com/26655166/89274341-131b5c80-d641-11ea-8b26-51385319f6a4.png">